Gm jscope improvements

GabrieleManduchi opened the pull request:
This version implements the features agreed in September meeting, namely:
- Implementation of a brand new LocalDataProvider in a pure Java implementation using pipe communication with a subprocess running mdsip-client-local
- Following the same approach a new SSHDataProvider has been added. 
- Added Command line Data Provider selection: -s <mdsip address> forces the use of MdsDataProvider connected to the specified mdsip server; -l forces the use of LocalDataProvider; -ssh <ssh address> forces the use of SSHDataProvider. In the above bases, the Network menu bar item is disabled, otherwise it is enabled and Data Server selection is performed in the usual way. NOTE: when a configuration file is passed as argumen and either -l -s or -ssh are defines, the data server specified in the configuration file is ignored.
- Made jScope compatible with DWScope configurations

GabrieleManduchi committed:
Remove useless test

GabrieleManduchi committed:
Remove useless test

joshStillerman committed:
Feature: add support for D-TACQ ACQ2106 with 1 to 6 ACQ423 modules

joshStillerman committed:
Fix: Add new device implimentation for acq2106_423_xst

GabrieleManduchi committed:
Feature:New LocalDataProvider

New pure Java implementaion of LocalDataProvider using local mdsip access and Process creation and pipe management directly in Java

tfredian committed:
Fox: no directories in Debian package checks

GabrieleManduchi committed:
Feature:Clean jScope sources and make jScope able to read DWScope configurations

Several jScope classes have been cleaned, and the management of label declarations (now converted to  y_label) in DWScope configuration files has been added to achieve backward compatibility.

GabrieleManduchi committed:
Merge branch 'gm-jscope-improvements' of https://github.com/MDSplus/mdsplus into gm-jscope-improvements

tfredian commented:
This javascope build failed. See: http://jenkins2.mdsplus.org/job/1-pull-request-test/lastFailedBuild/artifact/make_jars.log/*view*/
I assume you forgot to add the MdsConnectionLocal.java module in your commits.

If you add the MdsConnectionLocal.java to your branch you should be able to push it which will trigger a new build.

I'll look into fixing the build.sh script so that failures to build the jars will not trigger all the platform builds.

GabrieleManduchi committed:
Fix:SaveData Provider in Configuration file

When DataProvider was passed in command line, it was not saved in configuration file on "Save Configuration"

GabrieleManduchi committed:
Fix:Add forgotten files

Added new files jScope/MdsConnectionLocal.java jScope/MdsConnectionSSH.java jScope/SSHDataProvider.java

tfredian reviewed (approved):
Looks ok but not sure how you got hts commits in it. Rebasing on alpha looks fine though.

tfredian commented:
Replaced by: https://github.com/MDSplus/mdsplus/pull/1413
